File: android/src/oldarch/com/zoontek/rnbootsplash/RNBootSplashModule.java
package com.zoontek.rnbootsplash;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.bridge.LifecycleEventListener;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.module.annotations.ReactModule;

import java.util.Map;

@ReactModule(name = RNBootSplashModuleImpl.NAME)
public class RNBootSplashModule extends ReactContextBaseJavaModule implements LifecycleEventListener {

  public RNBootSplashModule(ReactApplicationContext reactContext) {
    super(reactContext);
    reactContext.addLifecycleEventListener(this);
  }

  @NonNull
  @Override
  public String getName() {
    return RNBootSplashModuleImpl.NAME;
  }

  @Nullable
  @Override
  public Map<String, Object> getConstants() {
    return RNBootSplashModuleImpl.getConstants(getReactApplicationContext());
  }

  @Override
  public void onHostDestroy() {}

  @Override
  public void onHostResume() {
    RNBootSplashModuleImpl.onHostResume();
  }

  @Override
  public void onHostPause() {
    RNBootSplashModuleImpl.onHostPause();
  }

  @ReactMethod
  public void hide(boolean fade, Promise promise) {
    RNBootSplashModuleImpl.hide(getReactApplicationContext(), fade, promise);
  }

  @ReactMethod
  public void isVisible(Promise promise) {
    RNBootSplashModuleImpl.isVisible(promise);
  }
}

File: android/src/oldarch/com/zoontek/rnbootsplash/RNBootSplashModule.kt
package com.zoontek.rnbootsplash

import com.facebook.react.bridge.LifecycleEventListener
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.module.annotations.ReactModule

@ReactModule(name = RNBootSplashModuleImpl.NAME)
class RNBootSplashModule(reactContext: ReactApplicationContext) :
  ReactContextBaseJavaModule(reactContext), LifecycleEventListener {

  init {
    reactContext.addLifecycleEventListener(this)
  }

  override fun getName(): String {
    return RNBootSplashModuleImpl.NAME
  }

  override fun onHostResume() {}

  override fun onHostPause() {}

  override fun onHostDestroy() {
    RNBootSplashModuleImpl.onHostDestroy()
  }

  override fun getConstants(): Map<String, Any> {
    return RNBootSplashModuleImpl.getConstants(reactApplicationContext)
  }

  @ReactMethod
  fun hide(fade: Boolean, promise: Promise) {
    RNBootSplashModuleImpl.hide(reactApplicationContext, fade, promise)
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun isVisible(): Boolean {
    return RNBootSplashModuleImpl.isVisible()
  }
}
